app/src/main/kotlin/com/hippo/ehviewer/ui/reader/SliderPagerDoubleSync.kt

```diff
@@ -0,0 +1,41 @@
+package com.hippo.ehviewer.ui.reader
+
+import androidx.compose.foundation.interaction.collectIsDraggedAsState
+import androidx.compose.foundation.lazy.LazyListState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.compose.runtime.snapshotFlow
+import kotlinx.coroutines.flow.collectLatest
+
+class SliderPagerDoubleSync(private val lazyListState: LazyListState) {
+    private var sliderFollowPager by mutableStateOf(true)
+    var sliderValue by mutableIntStateOf(0)
+
+    fun sliderScrollTo(index: Int) {
+        sliderFollowPager = false
+        sliderValue = index
+    }
+
+    @Composable
+    fun Sync() {
+        val fling by lazyListState.interactionSource.collectIsDraggedAsState()
+        if (fling) sliderFollowPager = true
+        if (sliderFollowPager) {
+            LaunchedEffect(Unit) {
+                snapshotFlow { lazyListState.layoutInfo }.collect { info ->
+                    sliderValue = info.visibleItemsInfo.last().index
+                }
+            }
+        } else {
+            LaunchedEffect(Unit) {
+                snapshotFlow { sliderValue }.collectLatest { index ->
+                    lazyListState.animateScrollToItem(index)
```

Comment:
`sliderValue` is 1-based while lazy list's index is 0-based.

Reply:
Done

Reply:
Done